components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -747,6 +755,31 @@ private void writeToFile(URI parentName, String newFileName, String content) thr
         }
     }
 
+    /**
+     * Writes metadata related to the given resource.
+     *
+     * @param parent parent dir of the resource
+     * @param resourceFileName filename of the resource
+     * @param metadata metadata properties object
+     * @throws SynapseException
+     */
+    private void writeMetadata(File parent, String resourceFileName, Properties metadata) throws SynapseException {
```

Comment:
There are some duplicate code in this method with `writeToFile` method. Shall we refactor?